### PR TITLE
Polly

### PR DESCRIPTION
**Pull request by thewilsonator**
Enable linking with Polly, the polyhedral optimiser. There are no tests yet, this is very much an experiment.

### CODE REVIEW

**Comment by JohanEngelen** [comment]
See https://github.com/ldc-developers/ldc/issues/1614

**Comment by thewilsonator** [comment]
Thanks for the pointers. Implementation is there, not sure where in the pipeline to put the passes yet.

File: gen/uda.cpp [hunk 1]
```diff
@@ -393,8 +393,9 @@ void applyFuncDeclUDAs(FuncDeclaration *decl, IrFunction *irFunc) {
       applyAttrSection(sle, func);
     } else if (ident == Id::udaTarget) {
       applyAttrTarget(sle, func);
-    } else if (ident == Id::udaWeak || ident == Id::udaKernel) {
+    } else if (ident == Id::udaWeak || ident == Id::udaKernel || Id::udaPolly) {
       // @weak and @kernel are applied elsewhere
+      // @polly is an inverted attribute and is applied elsewhere
```
Comment (JohanEngelen) [inline gen/uda.cpp]: i think i know what you mean but "inverted" will need a better explanation ;-)

**Comment by thewilsonator** [comment]
The clone time is pretty bad, but I think the link time is what is really killing it.

**Comment by thewilsonator** [comment]
I uploaded a debug build, that'd be why.

**Comment by thewilsonator** [comment]
stuffed up with git. superseded by https://github.com/ldc-developers/ldc/pull/2243

**Pull request closed without merging**